colour/volume/tests/test_spectrum.py

```diff
@@ -243,9 +243,9 @@ def test_nan_is_within_visible_spectrum(self):
         """
 
         cases = [-1.0, 0.0, 1.0, -np.inf, np.inf, np.nan]
-        cases = set(permutations(cases * 3, r=3))
-        for case in cases:
```

Comment:
One reason for the iteration case by case was that it is easier to figure out when the tested definition breaks. It is slower that being said, will need to think about it as we also have plenty of them.

Reply:
This uses a *huge* amount of time (I think because of setup in the underlying code?) I know the old adage is that machine time is *free-ish* and developer time is very expensive but this seems much to excessive to me. Besides, I like running tests. And hate waiting too long. 

If it fails, then presumably developers are going to enter this function anyway and sort out the failing case very quickly.